AtUnpack/AtGRAWUnpacker.cxx

```diff
@@ -233,15 +234,35 @@ void AtGRAWUnpacker::ProcessFile(Int_t coboIdx)
                for (Int_t iTb = 0; iTb < 512; iTb++)
                   pad->SetRawADC(iTb, rawadc[iTb]);
 
-               Int_t fpnCh = GetFPNChannel(iCh);
-               Double_t adc[512] = {0};
-               fPedestal[coboIdx]->SubtractPedestal(512, frame->GetSample(iAget, fpnCh), rawadc, adc,
-                                                    fFPNSigmaThreshold);
+               if (fIsSubtractFPN) {
+                  Int_t fpnCh = GetFPNChannel(iCh);
+                  Double_t adc[512] = {0};
+                  fPedestal[coboIdx]->SubtractPedestal(512, frame->GetSample(iAget, fpnCh), rawadc, adc,
+                                                       fFPNSigmaThreshold);
 
-               for (Int_t iTb = 0; iTb < 512; iTb++)
-                  pad->SetADC(iTb, adc[iTb]);
+                  for (Int_t iTb = 0; iTb < 512; iTb++)
+                     pad->SetADC(iTb, adc[iTb]);
 
-               pad->SetPedestalSubtracted(kTRUE);
+                  pad->SetPedestalSubtracted(kTRUE);
+               }
+
+               if (fIsSaveLastCell) {
+                  auto lastCellPad =
+                     dynamic_cast<AtPadValue *>(pad->AddAugment("lastCell", std::make_unique<AtPadValue>()));
+                  lastCellPad->SetValue(frame->GetLastCell(iAget));
+               }
+            }
+            if (fSaveFPN) {
+               Int_t fpnMap[4] = {11, 22, 45, 56};
```

Comment:
Don't hard code FPN values. Get them from the mapping.

Reply:
This will require adding a new function to the AtMap class with something like this signature
`std::vector<AtPadReference> GetFPNchannels(int cobo=0, int asad=0, int aget = 0)`